src/js/utils/index.ts

```diff
@@ -2,23 +2,64 @@ import { mapKeys, isEqual } from 'lodash';
 import { DataDisplaySettingsReducerState } from '../reducers/datadisplaysettings';
 import { VtState } from '../types';
 
-export function renameData(rawFrameData, moveNameType, inputNotationType) {
-
-  let renameKey = "";
-  if (moveNameType === "official") {
-    renameKey = "moveName"
-  } else if (moveNameType === "common") {
-    renameKey = "cmnName";
-  } else if (moveNameType === "inputs" && inputNotationType) {
-      renameKey = inputNotationType;
+/**
+ * Renames the moves in the character frame data to reflect the user's desired naming convention
+ * @param {string} rawFrameData The frame data for the current character
+ * @param {DataDisplaySettingsReducerState} dataDisplayState The Redux state containing various move text render settings
+ * @returns The frame data JSON object with renamed moves
+ */
+export function renameData(rawFrameData, dataDisplayState: DataDisplaySettingsReducerState) {
+  const renameFrameData = (rawData, renameKey, notationDisplay) => {
+    switch (notationDisplay) {
+      case "fullWord":
+        return mapKeys(rawData, (moveValue, moveKey) => moveValue[renameKey] ? moveValue[renameKey] : moveKey);
+      case "shorthand":
+        return renameFrameDataToShorthand(rawData);
+      default:
+        break;
+    }
   }
 
-  const renamedFrameData = mapKeys(rawFrameData,
-    (moveData, moveKey) =>
-      moveData[renameKey] ? moveData[renameKey] : moveKey
-  );
+  const renameFrameDataToShorthand = (rawData) => {
+    let rename = mapKeys(rawData, (moveValue, moveKey) => {
+      if (moveValue.moveType === "normal" && !moveValue.movesList) {
+        return formatMoveName(moveValue);
+      } else {
+        return moveKey;
+      }
+    });
+
+    return rename;
+  }
+
+  const formatMoveName = (moveData) => {
+    let truncatedMoveName: string = "";
+    const numberToDirectionMap: Map<string, string> = new Map([
```

Comment:
For this part, I'm looking at the `numCmd` property and making a string building decision based on the number I "see" after I split the command. I'm not modifying the `numCmd` itself for inputs, and the change for line 59 prevents the previous blanket modification if the user wants to see inputs and they leave "shorthand" selected.

From my viewpoint, looking at the `numCmd` requires a little less thought for coalescing as opposed to looking at the `moveName` property, splitting it, making a decision based on the length of the array (i.e., accounting for Neutral Jump [input]), then making the string modifications.

I _can_ change this implementation to leverage the `moveName` property process I detailed if you would rather see that in the codebase. I'd still use a map because it eliminates chunky if-else statements, but my take is that having something like 

```js
const CONTAINS_NEUTRAL_JUMP = 3

let splitMoveName: string[] = moveData.moveName.split(' ');
if (splitMoveName.length === CONTAINS_NEUTRAL_JUMP) {
// ...
}
``` 

is logically correct, it isn't as clear when reading it the first time.

Reply:
No, that's totally nice. Keep it as is so!